Common/Orders/Order.cs

```diff
@@ -187,6 +205,83 @@ protected Order(string symbol, SecurityType type, int quantity, OrderType order,
         }
 
         /// <summary>
+        /// Create cancel request.
+        /// </summary>
+        /// <returns></returns>
+        public CancelOrderRequest CancelRequest()
+        {
+            return new CancelOrderRequest
+            {
+                Id = Guid.NewGuid(),
+                OrderId = Id,
+                Created = DateTime.Now
+            };
+        }
+
+        /// <summary>
+        /// Factory method to create a new order from submit request.
+        /// </summary>
+        /// <param name="source"></param>
+        /// <returns></returns>
+        public static Order Create(SubmitOrderRequest source)
+        {
+            switch (source.Type)
+            {
+                case OrderType.Limit:
+                    return new LimitOrder(source);
+                case OrderType.Market:
+                    return new MarketOrder(source);
+                case OrderType.MarketOnClose:
+                    return new MarketOnCloseOrder(source);
+                case OrderType.MarketOnOpen:
+                    return new MarketOnOpenOrder(source);
+                case OrderType.StopLimit:
+                    return new StopLimitOrder(source);
+                case OrderType.StopMarket:
+                    return new StopMarketOrder(source);
+                default:
+                    throw new NotSupportedException(string.Format("{0} OrderType is not supported.", source.Type));
+            }
+        }
+
+        /// <summary>
+        /// Apply update to the order.
+        /// </summary>
+        /// <param name="request"></param>
+        public virtual void ApplyUpdate(UpdateOrderRequest request)
+        {
+            this.Time = request.Created;
+            Quantity = request.Quantity;
+            Tag = request.Tag;
+        }
+
+        /// <summary>
+        /// Copy order for update operations and messaging.
+        /// </summary>
+        /// <returns></returns>
+        public abstract Order Copy();
```

Comment:
Please update this method to be named `Clone` to match the pattern established with `BaseData`